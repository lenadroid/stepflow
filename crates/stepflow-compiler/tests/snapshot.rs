use std::fs::File;
use std::io::BufReader;

use stepflow_compile::compile;
use stepflow_steps::Plugins;

#[test]
fn inputs_snapshot() {
    let plugins = Plugins::new();

    insta::glob!("inputs/*.yaml", |path| {
        let file = File::open(path).unwrap();
        let reader = BufReader::new(file);

        let flow = stepflow_workflow::Flow::from_yaml_reader(reader).unwrap();
        let result = compile(&plugins, flow).map_err(|e| e.to_string());

        insta::assert_yaml_snapshot!(result)
    })
}