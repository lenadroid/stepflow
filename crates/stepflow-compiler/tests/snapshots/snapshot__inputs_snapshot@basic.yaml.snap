---
source: crates/stepflow-compiler/tests/snapshot.rs
expression: result
input_file: crates/stepflow-compiler/tests/inputs/basic.yaml
---
Err: No plugin found
